test/irmin/test_type.ml

```diff
@@ -226,21 +226,21 @@ let test_to_string () =
   in
 
   (* Test cases for basic types *)
-  test "unit" T.unit () "";
+  test "unit" T.unit () "()";
   test "bool{true}" T.bool true "true";
   test "bool{false}" T.bool false "false";
-  test "char" T.char 'a' "a";
+  test "char" T.char 'a' "'a'";
   test "int" T.int (-100) "-100";
   test "int32" T.int32 Int32.max_int "2147483647";
   test "int64" T.int64 Int64.max_int "9223372036854775807";
   test "float" T.float (-1.5) "-1.5";
   test "float{NaN}" T.float Stdlib.nan "nan";
-  test "bytes" T.bytes (Bytes.make 5 'a') "aaaaa";
-  test "string" T.string "foo\nbar\\" "foo\nbar\\";
+  test "bytes" T.bytes (Bytes.make 5 'a') "\"aaaaa\"";
+  test "string" T.string "foo\nbar\\" "\"foo\\nbar\\\\\"";
```

Comment:
I think escaping strings to be directly importable to OCaml is probably a step too far: I think it would just confuse me. Other thoughts? (CC @samoht)

Reply:
yea, no need to espace strings I think

Reply:
Actually after some thoughts I am not so sure. If the goal is to have an OCaml output (e.g. something similar to `Fmt.Dump`) better stay consistent.

Reply:
The problem from my perspective is that `Type.to_string` is directly derived from this pretty-printer, and I wouldn't expect such behaviour from a `to_string` function. Perhaps the answer is to have an optional `?escape_ocaml` that is set to false in the `to_string` case.

Having thought of that, this change probably introduces a large number of cases where `to_string >> of_string` is always an Error (since `of_string` is still delegating to `Type_json.of_string` in most complex cases.) I think that probably needs to be dealt with in one way or another.

Reply:
I feel like not escaping would prove frustrating quite quickly–imagine having an "almost-ready-to-paste" representation, but still having to quote manually before you can actually paste. As for the "it would confuse me", I guess it really depends on context. A standalone string with quotes might be confusing, but a quoted string in the inside an option or a record feels natural enough to me.

Reply:
Also, if we don't escape, we might get confused if we got something an output like `Some`, which could just as much be the constructor or the string `"Some"`.

Reply:
Fair points. The case for Bytes still applies since there is no bytes literal (and we're currently making them look like string literals...). Can we go back to the default behaviour there?

Reply:
> Also, if we don't escape, we might get confused if we got something an output like `Some`, which could just as much be the constructor or the string `"Some"`.

I think this confuses surrounding with quote marks (which I think is definitely good) with escaping the inner string (which I think is probably not, or is at least only useful in certain contexts). Again, I think it's fine for `pp` to do something like that, but definitely wouldn't expect it of `to_string`.

Reply:
Oh, there might actually be a significant issue that I missed. After searching for `to_string` in the codebase quickly, it seems like the pretty-printer is used in a lot of places as a quick way to turn an "unknown" Irmin type into an unquoted string (e.g. in `irmin-git`, in `dot.ml` or in `commit.ml`).

What is `Irmin.Type.to_string` actually supposed to be used for, then?

Reply:
> > Also, if we don't escape, we might get confused if we got something an output like `Some`, which could just as much be the constructor or the string `"Some"`.
> 
> I think this confuses surrounding with quote marks (which I think is definitely good) with escaping the inner string (which I think is probably not, or is at least only useful in certain contexts). Again, I think it's fine for `pp` to do something like that, but definitely wouldn't expect it of `to_string`.

Right now, `to_string` and `pp` are defined using one another (`let to_string ty = Fmt.to_to_string (pp ty)`).
Should we break this behavior, and separate the two?

Reply:
(As said above, I think the solution is to have `pp` take `?(escape_ocaml = true)` which is set to `false` when building the `to_string` function. Then `of_string` will need to implement the inverse of `pp ~escape_ocaml:true`.)

Reply:
Would `escape_ocaml=false` still output quotes around strings, or would we get rid of them altogether to maintain backwards compatibility?

Reply:
I vote not, so that `Type.(to_string string)` is identity, but there are advantages both ways. I leave it up to you :slightly_smiling_face: